extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeChannelExtractor.java

```diff
@@ -158,53 +161,105 @@ public String getName() throws ParsingException {
         }
 
-        return YoutubeParsingHelper.fixThumbnailUrl(avatarJsonObjectContainer.getObject("avatar")
-                .getArray("thumbnails")
-                .getObject(0)
-                .getString("url"));
+        return channelHeader.map(header -> {
+            switch (header.headerType) {
+                case PAGE:
+                    return header.json.getObject("content")
+                            .getObject("pageHeaderViewModel")
+                            .getObject("image")
+                            .getObject("contentPreviewImageViewModel")
+                            .getObject("image")
+                            .getArray("sources")
+                            .getObject(0)
+                            .getString("url");
+
+                case INTERACTIVE_TABBED:
+                    return header.json.getObject("boxArt")
+                            .getArray("thumbnails")
+                            .getObject(0)
+                            .getString("url");
+
+                case C4_TABBED:
+                case CAROUSEL:
+                default:
+                    return header.json.getObject("avatar")
+                            .getArray("thumbnails")
+                            .getObject(0)
+                            .getString("url");
+            }
+        })
+                .map(YoutubeParsingHelper::fixThumbnailUrl)
+                .orElseThrow(() -> new ParsingException("Could not get avatar URL"));
     }
 
     @Override
     public String getBannerUrl() throws ParsingException {
+        assertPageFetched();
         if (channelAgeGateRenderer != null) {
-            return "";
+            return null;
         }
 
-        return getChannelHeaderJson().flatMap(header -> Optional.ofNullable(
-                        header.getObject("banner")
-                                .getArray("thumbnails")
-                                .getObject(0)
-                                .getString("url")))
-                .filter(url -> !url.contains("s.ytimg.com") && !url.contains("default_banner"))
-                .map(YoutubeParsingHelper::fixThumbnailUrl)
-                // Channels may not have a banner, so no exception should be thrown if no banner is
-                // found
-                // Return null in this case
-                .orElse(null);
+        if (channelHeader.isPresent()) {
+            final ChannelHeader header = channelHeader.get();
+            if (header.headerType == HeaderType.PAGE) {
+                // No banner is available on pageHeaderRenderer headers
+                return null;
+            }
+
+            return Optional.ofNullable(header.json.getObject("banner")
+                    .getArray("thumbnails")
+                    .getObject(0)
+                    .getString("url"))
+                    .filter(url -> !url.contains("s.ytimg.com") && !url.contains("default_banner"))
```

Comment:
Why are you not returning a banner at all if the url contains `s.ytimg.com` or `default_banner`? This seems to date back to the initial commit of the extractor: [see here](https://github.com/AudricV/NewPipeExtractor/blame/7581c1200b7795d8fbb0769930f5981a962e347b/services/youtube/YoutubeChannelExtractor.java#L142)

Reply:
I wanted to keep the original behavior, that's the reason I didn't change this. Like you said, that's indeed useless.